superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterSets/FilterSetUnit.tsx

```diff
@@ -41,36 +40,38 @@ const IconsBlock = styled.div`
 `;
 
 type FilterSetUnitProps = {
-  filters: Filter[];
   editMode?: boolean;
   isApplied?: boolean;
   filterSet?: FilterSet;
   filterSetName?: string;
-  dataMaskApplied?: DataMaskUnit;
+  dataMaskSelected?: DataMaskUnit;
   setFilterSetName?: (name: string) => void;
   onDelete?: HandlerFunction;
   onEdit?: HandlerFunction;
+  onInvalidate?: HandlerFunction;
 };
 
 const FilterSetUnit: FC<FilterSetUnitProps> = ({
-  filters,
   editMode,
   setFilterSetName,
   onDelete,
   onEdit,
   filterSetName,
-  dataMaskApplied,
+  dataMaskSelected,
   filterSet,
   isApplied,
+  onInvalidate,
 }) => {
   const menu = (
     <Menu>
       <Menu.Item onClick={onEdit}>{t('Edit')}</Menu.Item>
+      <Menu.Item onClick={onInvalidate}>{t('Invalidate')}</Menu.Item>
```

Comment:
I think "Invalidate" might be a misleading name - I'd propose removing this option if the metadata is ok, and calling it "Rebuild", "Refresh" or something similar with a tooltip that explains that it will remove invalid filters/settings.